hw/ip/csrng/doc/_index.md

```diff
@@ -91,3 +91,208 @@ This document aims to make the distinction between physical entropy and factory
+- `xcnt` : transfer count - sets the total length of the transfer in beats, where zero is 1 beat
+- `flgs` : flags - used to allow for additional features
+- `creq` : count request - used only for the `generate` command, sets the number of genbits bits to be recieved
+
+
+
+                                                
+#### Command Header                                
+The application command header is 32 bits wide. Below is a description of the fields of this header:
+
+```
+  31          23          15            7             0
+  +-----------+-----------+------+------+------+------+
+  +          creq         | flgs | xcnt | shid | acmd |
+  +-----------+-----------+------+------+------+------+
+
+
+```
+
+
+- Total header: bits 31:0
+    - acmd: bits 3-0 
+        - 4'h0: invalid command
+        - 4'h1: instantiate command
+        - 4'h2: reseed command  
+        - 4'h3: generate command
+        - 4'h4: update command
+        - 4'h5: uninstantiate command
+        - 4'h6-4'hF: reserved
+    - shid: bits 7-4
+        - supports up to 16 state handle ids
+    - xcnt: bits 11-8
+        - Number of words (32b/word) to transfer into csrng
+    - flgs: bits 15-11
+        - flags for features
+    - creq: bits 31-16
+        - Number of 128b genbits words (128b/word) requested
+
+
+#### Specific Command Examples
+
+- instantiate:
+    - acmd: 4'h1 - encoded command
+    - shid: 4'h3 - reserved by csrng, set by the app number
+    - xcnt: 4'hc - a count of up to 12 words (32b/word) of seed or additional data 
+    - flgs: 4'h1 - in this case, bit 12 set is full deterministic mode
+      full_deterministic_mode: if set, use only app seed, otherwise
+      use entropy and app seed as personalization string
+
+- reseed:
+    - acmd: 4'h2 - encoded command
+    - shid: 4'h3 - reserved by csrng, set by the app number
+    - xcnt: 4'h0 - a count of up to 12 words (32b/word) of seed or additional data
+    - flgs: 4'h0 - no flags sets
+
+- generate:
+    - acmd: 4'h3 - encoded command
+    - shid: 4'h3 - reserved by csrng, set by the app number
+    - xcnt: 4'h0 - a count of up to 12 words (32b/word) of seed or additional data
+    - flgs: 4'h1 - in this case, bit 12 set is prediction resistance flag
+    - creq: 16'h0020 - returns 32 words (128b/word) of generated bits on the genbits bus
+
+- update:
+    - acmd: 4'h4 - encoded command
+    - shid: 4'h3 - reserved by csrng, set by the app number
+    - xcnt: 4'h4 - a count of up to 12 words (32b/word) of seed or additional data
+    - flgs: 4'h0 - no flags sets
+
+- uninstantiate:
+    - acmd: 4'h5 - encoded command
+    - shid: 4'h3 - reserved by csrng, set by the app number
+    - xcnt: 4'h0 - always zero
+    - flgs: 4'h0 - no flags sets
+
+
+#### Waveforms
+
+##### Instantiate Request
+
+{{< wavejson >}}
+{signal: [
+   {name: 'clk'            , wave: 'p...............|.....'},
+   {name: 'csrng_req_vld'  , wave: '01............0.|.....'},
```

Comment:
Nit: Most of our existing ready/valid signals are actually using `_valid`, `_ready`, and `_data` as suffix.

Reply:
It looks like @mwbranstad has taken care of this.